source/family/cypress/PSoC6xxx/PSOC6xxx.c

```diff
@@ -0,0 +1,1449 @@
+*  SWDPutData
+***************************************************************************//**
+* Calls the functions to write data segment on to the SWD lines and checks the
+* acknowledgement to decide whether the write was successful or not.
+*
+* @param[in]  pbData    Pointer to the address location where data to be written
+*                       to the SWD lines is stored.
+*
+******************************************************************************/
+static void SWDPutData(uint8_t  *pbData)
+{
+    uint8_t bParity;
+    uint8_t enableInterrupts;
+
+    enableInterrupts = CyEnterCriticalSection();
+    gbStatsReg = 0u;
+    bParity = SWDComputeDataParity(pbData);
+    SWDPutHeaderSegment();
+
+    SWDClockTrn();
+    SWDGetAckSegment();
+    gbAck = ack0 | (ack1 << 1u) | (ack2 << 2u);
+    SWDClockTrn();
+
+    switch (gbAck)
+    {
+        case SWD_ACK:
+        {
+            SWDPutDataSegment(pbData);
+
+            if (bParity != 0u)
+            {
+                SWD_SET_SDA_HI;
+            }
+            else
+            {
+                SWD_SDA = 0u;
+            }
+
+            /* 46th clock ends SWD transaction */
+            SWD_CLOCK_BIT;
+
+            /* keep Data low to not start next SWD transaction on 47th clock */
+            SWD_SDA = 0u;
+            SWD_CLOCK_BIT; /* 47th clock is required to move Addr/Data from DAP to AHB bus. */
+
+            SWD_CLOCK_BIT; /* 48th clock CDT 127910 */
+            SWD_CLOCK_BIT; /* 49th clock */
+            SWD_CLOCK_BIT; /* 50th clock */
+            SWD_CLOCK_BIT; /* 51th clock */
+            SWD_CLOCK_BIT; /* 52th clock */
+            SWD_CLOCK_BIT; /* 53th clock */
+            SWD_CLOCK_BIT; /* 54th clock  CDT 127910 */
+
+            SWD_SET_SDA_IN;
+            SWD_SET_SDA_HI;
+            SWD_SET_SCK_HI;
+
+            gbStatsReg |= gbAck;
+            break;
+        }
+
+        case SWD_WAIT:
+        case SWD_FAULT:
+        {
+            SWD_SET_SDA_IN;
+            SWD_SET_SDA_HI;
+            SWD_SET_SCK_HI;
+            gbStatsReg |= gbAck;
+            break;
+        }
+
+        default:
+        {
+            SWD_SET_SDA_IN;
+            SWD_SET_SDA_HI;
+            SWD_SET_SCK_HI;
+
+            /* Protocol Error - no response from target */
+            gbStatsReg |= SWD_ERROR;
+            break;
+        }
+    }
+    CyExitCriticalSection(enableInterrupts);
+  return;
+}
+
+/******************************************************************************
+*  SWDWrite
+***************************************************************************//**
+* Executes SWD-write command on the bus. Command located at bufOut,
+* result (ACK) is returned to buf_ACK buffer.
+*
+* @param[in]  bufOut    pointer to buffer
+* @param[out] buf_ACK   buffer for return (ACK).
+*
+******************************************************************************/
+static void SWDWrite(uint8_t *bufOut, uint8_t *buf_ACK)
```

Comment:
Can you make use of the existing SWD functions in `source/daplink/cmsis_dap` and `source/daplink/interface/swd_host.*`? We really need to not have two separate SWD implementations in the code.

Reply:
I switched to your proposed implementation.